torch/onnx/symbolic_opset17.py

```diff
@@ -54,3 +55,128 @@ def layer_norm(
         epsilon_f=eps,
         axis_i=axis,
     )
+
+
+@_onnx_symbolic("aten::stft")
+@symbolic_helper.parse_args("v", "i", "i", "i", "v", "b", "b", "b")
+@_beartype.beartype
+def stft(
+    g: jit_utils.GraphContext,
+    input: _C.Value,
+    n_fft: int,
+    hop_length: Optional[int] = None,
+    win_length: Optional[int] = None,
+    window: Optional[_C.Value] = None,
+    normalized: bool = False,
+    onesided: Optional[bool] = True,
+    return_complex: Optional[bool] = False,
+) -> _C.Value:
+    """Associates `torch.stft` with the `STFT` ONNX operator.
+    Note that torch.stft calls _VF.stft, without centering or padding options.
+    Hence, this function does not contain these two arguments.
+    See torch.stft source code for more info.
+
+    Parameters
+    ----------
+    g : jit_utils.GraphContext
+        Graph to write the ONNX representation into
+    input : _C.Value
+        Input tensor for the transformation
+    n_fft : int
+        FFT size
+    hop_length : int, optional
+        Size of the hop. Defaults to floot(n_fft // 4)
+    win_length : int, optional
+        Size of the analysis window. Defaults to `n_fft`
+    window : _C.Value, optional
+        Analysis window. Defaults to a window of all ones.
+    normalized : bool, optional
+        Whether to return a normalized STFT.
+    onesided : bool, optional
+        Whether to return only half (+1) of the results, given the symmetry
+        of the STFT
+    return_complex : bool, optional
+        Whether to return the complex value (Note: Must be False or None)
+
+    Returns
+    -------
+    op
+        Operator for torch.stft associated with STFT (ONNX).
+    """
+    # Checks
+    assert (
+        return_complex is None or not return_complex
+    ), "STFT does not currently support complex types"
+
+    # Get STFT sizes
+    frame_step = hop_length if hop_length is not None else n_fft // 4
+    o_frame_step = g.op("Constant", value_t=torch.tensor(frame_step, dtype=torch.int64))
+    o_frame_length = g.op("Constant", value_t=torch.tensor(n_fft, dtype=torch.int64))
+
+    # Pre-process input if needed
+    o_signal = input
+    signal_rank = symbolic_helper._get_tensor_rank(o_signal)
+    if signal_rank == 1:
+        # Add batch dimension
+        o_signal = g.op(
+            "Unsqueeze",
+            o_signal,
+            g.op("Constant", value_t=torch.tensor([0], dtype=torch.int64)),
+        )
+    elif signal_rank > 2:
+        raise RuntimeError(
+            "STFT can only take inputs of 1 [signal] or 2 " "[batch, signal] dimensions"
+        )
+
+    # Get window and make sure it's the same size as the FFT size
+    o_window = window
+    n_win = symbolic_helper._get_tensor_dim_size(o_window, dim=0)
+    if n_win is not None:
+        assert n_win == n_fft, "Analysis window's size must the same as the FFT size"
```

Comment:
Would this ever happen with valid user inputs? If not, and assertion is good but may be useful to include that this is a bug etc. If it _can_ happen with valid user inputs and it's just a case we cannot support, then raise an error.

Reply:
Potentially suggest a fix in the error message as well, if possible, everywhere where errors are raised

Reply:
This only happens with non-valid user inputs. `torch.stft` also raises an error there, so I was mimicking that, not sure if it's necessary to leave it here. Happy to remove it if you think it's overkill.

I have updated all error messages with potential fixes.

Reply:
I see. If torch.stft already checks for it, I would go with an assert

Reply:
Done!